XenAdmin/Dialogs/AttachUsbDialog.cs

```diff
@@ -75,6 +96,31 @@ private void BuildList()
             }
         }
 
+        private void treeUsbList_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            UsbItem item = treeUsbList.SelectedItem as UsbItem;
+            if (item != null)
+            {
+                buttonAttach.Enabled = true;
+            }
+            else
+            {
+                buttonAttach.Enabled = false;
+            }
+        }
+
+        private void buttonAttach_Click(object sender, EventArgs e)
+        {
+            this.DialogResult = System.Windows.Forms.DialogResult.OK;

```

Comment:
The DialogResult can be set directly in the designer and there is no need to call Close() here

Reply:
Got it. Thanks.